lib/provider-registration/create-model.js

```diff
@@ -77,6 +77,16 @@ module.exports = function createModel ({ ProviderModel, koop, namespace }, optio
         }
       })
     }
+
+    async pullStream (req, callback) {
+      if (this.getStream) {
+        await this.before(req)
+        const providerStream = await this.getStream(req)
+        callback(null, providerStream)
```

Comment:
You need to add a try...catch here to make sure any error is sent back to the callback. See an example at the pull() function.

Yeah, it sucks to have promise and callback in the same place.